bertopic/_bertopic.py

```diff
@@ -1897,9 +1903,40 @@ def get_topic_tree(
 
         max_original_topic = hier_topics.Parent_ID.astype(int).min() - 1
 
+        # Prepare tree labels to print
+        child_left_ids = hier_topics.Child_Left_ID.astype(int)
+        child_right_ids = hier_topics.Child_Right_ID.astype(int)
+
+        # Get the new parent labels generated from `hierarchical_topics`
+        new_left_labels = {int(row["Child_Left_ID"]): row["Child_Left_Name"] for idx, row in hier_topics.iterrows()}
+        new_right_labels = {int(row["Child_Right_ID"]): row["Child_Right_Name"] for idx, row in hier_topics.iterrows()}
+
+        if custom_labels:
+            left_labels = {}
+            if isinstance(custom_labels, str):
+                for topic, kws_info in self.topic_aspects_[custom_labels].items():
+                    label = "_".join([kw[0] for kw in kws_info[:5]])  # displaying top 5 kws
+                    left_labels[topic] = label
```

Comment:
Should we add a check for when a user attempts to use a specific aspect but it is not found in `self.topic_aspects_`? I can imagine that a user (myself included) would mistype the name and the resulting error might not be entirely clear.

Reply:
This makes sense, I can add a check here. One thing I noticed when looking through how custom labels were handled in the plotting functions is that they all follow a pretty similar pattern:

https://github.com/MaartenGr/BERTopic/blob/510c15e0f3f3c6481f21087d9461e5fb5e8b89af/bertopic/plotting/_topics.py#L67
https://github.com/MaartenGr/BERTopic/blob/510c15e0f3f3c6481f21087d9461e5fb5e8b89af/bertopic/plotting/_heatmap.py#L99
https://github.com/MaartenGr/BERTopic/blob/510c15e0f3f3c6481f21087d9461e5fb5e8b89af/bertopic/plotting/_datamap.py#L123
https://github.com/MaartenGr/BERTopic/blob/510c15e0f3f3c6481f21087d9461e5fb5e8b89af/bertopic/plotting/_distribution.py#L62
https://github.com/MaartenGr/BERTopic/blob/510c15e0f3f3c6481f21087d9461e5fb5e8b89af/bertopic/plotting/_documents.py#L139

Maybe a more generic and reusable function can be created for this, which would check if the user specified aspect exists in `self.topic_aspects_`

Reply:
You're right, it makes no sense to do that check here but not at all of those other instances. For now, I'm alright either way. I can imagine wanting to stick with what I already use everywhere else for consistency as this might be a bit out of the scope of this particular PR. So it's up to you.